app/src/main/java/com/example/dragonx/viewmodel/RocketDetailsViewModel.kt

```diff
@@ -0,0 +1,82 @@
+package com.example.dragonx.viewmodel
+
+import android.util.Log
+import android.view.View
+import androidx.lifecycle.ViewModel
+import com.beust.klaxon.JsonArray
+import com.beust.klaxon.JsonObject
+import com.beust.klaxon.Parser
+import com.denzcoskun.imageslider.models.SlideModel
+import com.example.dragonx.models.Rocket
+
+import java.net.URL
+
+class RocketDetailsViewModel : ViewModel() {
+    var imageList = ArrayList<SlideModel>()
+
+
+    init {
+        Log.i("JsonParser", "JsonParser for RocketDetailsFragment was created")
+    }
+
+    suspend fun parseJson (rocketNumber:Int?): Rocket {
+        lateinit var rocket: Rocket
+        lateinit var jsonArray: JsonArray<JsonObject>
+
+        val result = URL("https://api.spacexdata.com/v4/dragons").readText()
+        val parser: Parser = Parser()
+        val stringBuilder: StringBuilder = StringBuilder(result)
+        jsonArray = parser.parse(stringBuilder) as JsonArray<JsonObject>
+        rocket = Rocket()
+        rocket.name = jsonArray.string("name")[rocketNumber!!]
+        rocket.description = jsonArray.string("description")[rocketNumber]
+        rocket.link = jsonArray.string("wikipedia")[rocketNumber]
+        rocket.height =
+            ("Diameter : meters : ${jsonArray.obj("diameter")[rocketNumber]?.double("meters")}," +
```

Comment:
What will you do if you will need to use height with another format? For example : " Diameter ${diameter}"